xbmc/filesystem/PluginDirectory.cpp

```diff
@@ -486,35 +485,26 @@ bool CPluginDirectory::WaitOnScriptResult(const std::string &scriptPath, int scr
       CScriptObserver scriptObs(scriptId, m_fetchComplete);
       if (!CGUIDialogBusy::WaitOnEvent(m_fetchComplete, 200))
       {
-        cancelled = true;
+        m_cancelled = true;
```

Comment:
this does not help if the plugin has opened another dialog like keyboard for input.